### ISSUE TITLE
Suggestion: document more tips for optimising browser builds

### ISSUE THREAD

**Issue by OliverJAsh** [mattiasw/ExifReader#131]
Thanks for this fantastic library!

I had a few struggles getting this library down to the right size when bundling for the browser via webpack.

I thought tree shaking would work its magic if I imported from the `src` folder (as directed by the `pkg.module` field), seeing as this folder uses ES Modules, but interestingly this ended up being much larger than a custom build (~20 KB gzipped compared to 3 KB gzipped). Do you know why that is? Perhaps we should add something to the documentation to advise against relying on just tree shaking.

Secondly, webpack will bundle the `buffer` module even though it's never used in the browser:

![image](https://user-images.githubusercontent.com/921609/99827497-9bbafb80-2b51-11eb-95d9-d6d7cc6b2125.png)

The fix is to add the following to webpack's config: `node: { Buffer: false }`. Here's a good description of what's going on here: https://github.com/webpack/webpack/issues/4240#issuecomment-510827283

After this fix:

![image](https://user-images.githubusercontent.com/921609/99827543-ad9c9e80-2b51-11eb-8791-647ccf0a162c.png)

Perhaps we can add this recommendation to the docs. WDYT?

**Comment by mattiasw** [comment]
Hi!

If I understand you correctly, I guess it's not made clear exactly what is happening when creating the custom build. As specified in the README it's the built file that is changed. Would it be clearer if this note was added to the custom build section?

```
NOTE: This only changes the built file (`exifreader/dist/exif-reader.js`),
not the source code. That means it's not possible to use the ES module (from the
`src` folder) or any tree-shaking to get the benefit of a custom build.
```

It would have been nice if it could work with importing the ES module directly but that seems a bit trickier. The source code itself would have to be manipulated instead of just the file outputted from the build. And that also adds the problem with having to store the original source code somewhere in case one would want to rebuild.

Regarding Buffer, thanks for pointing this out! I thought I had taken care of it here: https://github.com/mattiasw/ExifReader/blob/master/webpack.config.js#L82
But that will of course not help anyone importing from `src` directly. :-( Do you also get this behavior when using the built file? I'm wondering where in the README this should be mentioned.

**Comment by OliverJAsh** [comment]
> Would it be clearer if this note was added to the custom build section?

Yes that would be much clearer!

Intuitively I thought I would be able to achieve similar results to the custom build by just using `src` and then applying tree shaking (which works on ES Modules). However this did not seem to achieve the same results. I'm not sure why that is? Perhaps we could also add a warning that it is likely that the custom build will be smaller than using the ES modules version with tree shaking.



> But that will of course not help anyone importing from `src` directly. :-( Do you also get this behavior when using the built 

Yes I am consuming the built file in the `dist` folder. The `buffer` module is still bundled, not inside the `dist` file generated by webpack but inside _our own_ compiled code which is also generated by our version of webpack.

**Comment by mattiasw** [comment]
The reason tree shaking does not work is that there is no change in the actual source code when creating the custom build. So it will not know what parts should be excluded. The customization is done during the build of the `dist` file while transpiling the source files (by removing complete code paths, e.g. for PNG or XMP, but also removing the tags that are not used from a large dictionary). "Your" webpack will still think all the code is important and indeed used and nothing can be tree-shaken. I guess it would be possible to achieve some tree shaking by exposing other methods (like loadGPS or something). But it would probably be hard to even get close to the current custom build using that strategy,

Ah, yes, of course. I completely missed that angle with `Buffer`. I will add something.

**Comment by mattiasw** [comment]
Fixed in #134.
